File: 20. Right view of tree.cpp
#include <bits/stdc++.h>
using namespace std;
struct Node
{
    int data;
    Node* left;
    Node* right;
    Node(int d) : data(d), left(nullptr), right(nullptr) {}
};
void rightView(Node* root)
{
    if (!root)
        return;
    queue<Node*> q;
    q.push(root);
    while (!q.empty())
    {
        int sz = q.size();
        for (int i = 0; i < sz; i++)
        {
            Node* cur = q.front();
            q.pop();
            if (i == sz - 1)
                cout << cur->data << " ";
            if (cur->left)
                q.push(cur->left);
            if (cur->right)
                q.push(cur->right);
        }
    }
    cout << endl;
}
int main()
{
    Node* root = new Node(1);
    root->left = new Node(2);
    root->right = new Node(3);
    root->left->left = new Node(4);
    root->left->right = new Node(5);
    cout << "Right view: ";
    rightView(root);
    return 0;
}

File: 20. Right view of tree.py
from collections import deque
class Node:
    def __init__(self,data): self.data=data; self.left=None; self.right=None

def right_view(root):
    if not root: return
    q=deque([root])
    while q:
        sz=len(q)
        for i in range(sz):
            cur=q.popleft()
            if i==sz-1: print(cur.data, end=' ')
            if cur.left: q.append(cur.left)
            if cur.right: q.append(cur.right)
    print()

if __name__=='__main__':
    root=Node(1); root.left=Node(2); root.right=Node(3); root.left.left=Node(4); root.left.right=Node(5)
    print('Right view:', end=' '); right_view(root)
